tests/components/airvisual/test_config_flow.py

```diff
@@ -218,29 +244,43 @@ async def test_step_node_pro(hass):
         }
 
 
-async def test_step_import(hass):
-    """Test the import step for both types of configuration."""
+async def test_step_reauth(hass):
+    """Test that the reauth step works."""
     geography_conf = {
         CONF_API_KEY: "abcde12345",
         CONF_LATITUDE: 51.528308,
         CONF_LONGITUDE: -0.3817765,
     }
 
-    with patch(
-        "homeassistant.components.airvisual.async_setup_entry", return_value=True
-    ), patch("pyairvisual.api.API.nearest_city"):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_IMPORT}, data=geography_conf
-        )
+    MockConfigEntry(
+        domain=DOMAIN, unique_id="51.528308, -0.3817765", data=geography_conf
+    ).add_to_hass(hass)
 
-        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-        assert result["title"] == "Cloud API (51.528308, -0.3817765)"
-        assert result["data"] == {
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": "reauth"},
+        data={
```

Comment:
Instead of copying the same data, you can do something like 

```py
data={
    **DATA_USER,
    CONF_INTEGRATION_TYPE: INTEGRATION_TYPE_GEOGRAPHY
}
```